pkg/controller/deliverable/reconciler.go

```diff
@@ -260,6 +265,40 @@ func (r *Reconciler) trackDependencies(deliverable *v1alpha1.Deliverable, realiz
 	}
 }
 
+func (r *Reconciler) cleanupOrphanedObjects(ctx context.Context, previousResources, realizedResources []v1alpha1.RealizedResource) error {
+	log := logr.FromContextOrDiscard(ctx)
+
+	var orphanedObjs []*corev1.ObjectReference
+	for _, prevResource := range previousResources {
+		orphaned := true
+		for _, realizedResource := range realizedResources {
+			if realizedResource.StampedRef.Kind == prevResource.StampedRef.Kind &&
+				realizedResource.StampedRef.Name == prevResource.StampedRef.Name {
+				orphaned = false
+				break
+			}
+		}
+		if orphaned {
+			orphanedObjs = append(orphanedObjs, prevResource.StampedRef)
+		}
+	}
+
+	for _, orphanedObj := range orphanedObjs {
+		obj := &unstructured.Unstructured{}
+		obj.SetNamespace(orphanedObj.Namespace)
+		obj.SetName(orphanedObj.Name)
+		obj.SetGroupVersionKind(orphanedObj.GroupVersionKind())
+
+		log.V(logger.DEBUG).Info("deleting orphaned object", "object", orphanedObj)
+		err := r.Repo.Delete(ctx, obj)
+		if err != nil {
+			return err
```

Comment:
should we wrap this err? iirc, that's the pattern we use at least in most places

Reply:
No, because we end up ignoring it